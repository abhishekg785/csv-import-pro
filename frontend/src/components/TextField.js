// @flow

import React from 'react'
import { TextField as MuiTextField } from '@material-ui/core'

type Props = {
  value?: string,
  onChange: Function,
  classes: any,
}

function TextField({ value, classes, onChange }: Props) {
  return (
    <MuiTextField
      id="standard-search"
      label="Search"
      type="Search"
      value={value}
      onChange={onChange}
      className={classes}
    />
  )
}

TextField.defaultProps = {
  value: '',
}

export default TextField
