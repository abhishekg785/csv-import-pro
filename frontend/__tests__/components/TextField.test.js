import * as React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import TextField from '../../src/components/TextField'

configure({ adapter: new Adapter() })

describe('TextField', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<TextField />);
    expect(wrapper).toMatchSnapshot();
  })
})
