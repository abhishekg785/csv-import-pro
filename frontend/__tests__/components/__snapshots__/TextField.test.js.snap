// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": undefined,
      "id": "standard-search",
      "label": "Search",
      "onChange": undefined,
      "required": false,
      "select": false,
      "type": "Search",
      "value": "",
      "variant": "standard",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": undefined,
        "id": "standard-search",
        "label": "Search",
        "onChange": undefined,
        "required": false,
        "select": false,
        "type": "Search",
        "value": "",
        "variant": "standard",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TextField renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": undefined,
      "id": "standard-search",
      "label": "Search",
      "onChange": undefined,
      "required": false,
      "select": false,
      "type": "Search",
      "value": "",
      "variant": "standard",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": undefined,
        "id": "standard-search",
        "label": "Search",
        "onChange": undefined,
        "required": false,
        "select": false,
        "type": "Search",
        "value": "",
        "variant": "standard",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
